packages/nep141-erc20/src/natural-erc20/sendToNear/index.js

```diff
@@ -478,6 +478,16 @@ async function checkLock (transfer) {
 }
 
 async function checkSync (transfer) {
+  if (!transfer.checkSyncInterval) {
+    // checkSync every 20s: reasonable value to show the confirmation counter x/30
+    transfer = {
+      ...transfer,
+      checkSyncInterval: Number(process.env.sendToNearSyncInterval)
+    }
+  }
+  if (transfer.nextCheckSyncTimestamp && new Date() < new Date(transfer.nextCheckSyncTimestamp)) {
+    return transfer
+  }
```

Comment:
I agree with both comments. `checkSync` and transfer finalization functions (`unlock`, `mint`..) are usually called from the `@near-eth/client` via the `checkStatus()` loop and `act()`. So when `@near-eth/client` calls `act(transfer)` or `checkSync(transfer)`, it should only have to check the status of the transfer and let the connector library do the rest. On the other hand `initiate` is called by the frontend directly when creating a transfer (not by `@near-eth/client`). So the question is where to define connector specific parameters like the ether custodian address for example.

1. We could define a general bridge parameters object for all connectors in `@near-eth/client` and import it from connector libraries which would pick the parameters needed. Each connector function (`initiate`, `checkSync`, `unlock`...) can have an optional argument for bridge parameters so that they can be called standalone with custom parameters and without using `@near-eth/client`.
2. We could set parameters like in 1. but for each connector lib. This is very tedious for the user of our libraries.
3. We could define parameters at transfer creation and store them in the transfer object to reuse in next steps so that `@near-eth/client` doesn't track parameters. This is a problem if we need to change parameters while the transfer is in progress as the transfer will rely on old parameters.

I prefer 1. wdyt ?

Reply:
I like 1) the most. It is easier to reason about it, and to make new connectors in the future (i.e. NFT connector)